content/2_Bronze/Rect_Geo.mdx

```diff
@@ -164,72 +202,82 @@ int main() {
 <JavaSection>
 
 ```java
-
 import java.io.*;
 import java.util.*;
 
-public class billboard
-{
-	public static void main(String[] args) throws IOException
-	{
+public class Billboard {
+	private static final int MAX_POS = 2000;
+	public static void main(String[] args) throws IOException {
 		BufferedReader br = new BufferedReader(new FileReader("billboard.in"));
-		PrintWriter out = new PrintWriter(new FileWriter("billboard.out"));
-		int ok[][] = new int[2000][2000];
+		int visible[][] = new int[MAX_POS][MAX_POS];
 		for (int i = 0; i < 3; ++i) {
-			StringTokenizer st = new StringTokenizer(br.readLine());
-			int x1 = Integer.parseInt(st.nextToken());
-			int y1 = Integer.parseInt(st.nextToken());
-			int x2 = Integer.parseInt(st.nextToken());
-			int y2 = Integer.parseInt(st.nextToken());
-			x1 += 1000; y1 += 1000; x2 += 1000; y2 += 1000;
-			for (int x = x1; x < x2; ++x){
-				for (int y = y1; y < y2; ++y) {
-					if (i < 2) ok[x][y] = 1;
-					else ok[x][y] = 0;
+			StringTokenizer board = new StringTokenizer(br.readLine());
+			int x1 = Integer.parseInt(board.nextToken());
+			int y1 = Integer.parseInt(board.nextToken());
+			int x2 = Integer.parseInt(board.nextToken());
+			int y2 = Integer.parseInt(board.nextToken());
+			x1 += MAX_POS / 2;
+			y1 += MAX_POS / 2;
+			x2 += MAX_POS / 2;
+			y2 += MAX_POS / 2;
+			for (int x = x1; x < x2; x++) {
+				for (int y = y1; y < y2; y++) {
+					visible[x][y] = i < 2 ? 1 : 0;
 				}
 			}
 		}
+
 		int ans = 0;
-		for (int x = 0; x < 2000; ++x){
-			for (int y = 0; y < 2000; ++y){
-				ans += ok[x][y];
+		for (int x = 0; x < 2000; ++x) {
+			for (int y = 0; y < 2000; ++y) {
+				ans += visible[x][y];
 			}
 		}
+		PrintWriter out = new PrintWriter(new FileWriter("billboard.out"));
 		out.println(ans);
 		out.close();
 	}
 }
-
 ```
 
 </JavaSection>
 
 <PySection>
 
-Unfortunately, the naive Python solution only passes 9/10 test cases.
+Code runs marginally faster in Python when placed in a function, so we can use
+this to get all 10 test cases (When taken out of the function, the code passes
+only 9).
```

Comment:
This is good information to know, but it may be more confusing than helpful to people new to the subject. Also, saying that the naive solution doesn't work is perfectly fine since there's a better solution anyway.

Reply:
Well, the C++ and Java solution both work, so it's good for the Python solution to be consistent.

Reply:
Yes but it's not instructive in any way and has the potential to cause confusion, the way I see it.

Reply:
Not really. All I'm saying is that Python code has the potential to run faster when placed in a function. I can add an SO thread to help explain it.

Reply:
maybe link something for the "code runs marginally faster in Python when placed in a function" part?

Reply:
[Try this?](https://www.tutorialspoint.com/Why-does-Python-code-run-faster-in-a-function#:~:text=It%20is%20generally%20found%20that,we%20download%20from%20Python.org.)

Reply:
I was actually thinking of ![this thread here](https://stackoverflow.com/questions/11241523/why-does-python-code-run-faster-in-a-function).

Reply:
Both say pretty much the same thing, I think either one works